keyboards/mechwild/puckbuddy/puckbuddy.c

```diff
@@ -0,0 +1,263 @@
+                break;
+            case 7:
+                oled_write_P(PSTR("      DPI: 3200      "), false);
+                break;
+            case 8:
+                oled_write_P(PSTR("      DPI: 3600      "), false);
+                break;
+            case 9:
+                oled_write_P(PSTR("      DPI: 4000      "), false);
+                break;
+            default:
+                oled_write_P(PSTR("      DPI: ????      "), false);    // Should never display, here as a catchall
+        }
+        clear_screen = true;
+    } else {
+        if (clear_screen == true) {
+            oled_clear();
+            oled_render();
+            clear_screen = false;
+        }
+        render_logo();
+        oled_set_cursor(8,2);
+        switch (get_highest_layer(layer_state)) {
+            case 0:
+                oled_write_P(PSTR("Layer 0"), false);
+                break;
+            case 1:
+                oled_write_P(PSTR("Layer 1"), false);
+                break;
+            case 2:
+                oled_write_P(PSTR("Layer 2"), false);
+                break;
+            case 3:
+                oled_write_P(PSTR("Layer 3"), false);
+                break;
+            default:
+                oled_write_P(PSTR("Layer ?"), false);    // Should never display, here as a catchall
+        }
+        led_t led_state = host_keyboard_led_state();
+        oled_set_cursor(8,0);
+        oled_write_P(led_state.scroll_lock ? PSTR("SCRLK") : PSTR("     "), false);
+        oled_set_cursor(8,1);
+        oled_write_P(led_state.num_lock ? PSTR("NLCK ") : PSTR("     "), false);
+        oled_write_P(led_state.caps_lock ? PSTR("CAPS ") : PSTR("     "), false);
+        oled_set_cursor(8,3);
+        switch (keyboard_config.dpi_config) {   // 512, 1024, 2048, 3072, 4096
+            case 0:
+                oled_write_P(PSTR("DPI: 400 "), false);
+                break;
+            case 1:
+                oled_write_P(PSTR("DPI: 800 "), false);
+                break;
+            case 2:
+                oled_write_P(PSTR("DPI: 1200"), false);
+                break;
+            case 3:
+                oled_write_P(PSTR("DPI: 1600"), false);
+                break;
+            case 4:
+                oled_write_P(PSTR("DPI: 2000"), false);
+                break;
+            case 5:
+                oled_write_P(PSTR("DPI: 2400"), false);
+                break;
+            case 6:
+                oled_write_P(PSTR("DPI: 2800"), false);
+                break;
+            case 7:
+                oled_write_P(PSTR("DPI: 3200"), false);
+                break;
+            case 8:
+                oled_write_P(PSTR("DPI: 3600"), false);
+                break;
+            case 9:
+                oled_write_P(PSTR("DPI: 4000"), false);
+                break;
+            default:
+                oled_write_P(PSTR("DPI: ????"), false);    // Should never display, here as a catchall
+        }
+    }
```

Comment:
This might work better:
```suggestion
        oled_write_P(PSTR("DPI: "), false);
        oled_write(get_u16_str(dpi_array[keyboard_config.dpi_config], ' '), false);
```